src/windows/models/files_model.py

```diff
@@ -277,41 +442,84 @@ def update_file_thumbnail(self, file_id):
 
         # Refresh thumbnail for updated file
         self.ignore_updates = True
+        m = self.model
+
         if file_id in self.model_ids:
-            for row_num in range(self.model.rowCount()):
-                id_index = self.model.index(row_num, 5)
-                if file_id == self.model.data(id_index):
-                    # Update thumb for file
-                    thumb_index = self.model.index(row_num, 0)
-                    thumb_path = self.get_thumb_path(file_id, 1, clear_cache=True)
-                    item = self.model.itemFromIndex(thumb_index)
-                    item.setIcon(QIcon(thumb_path))
-                    item.setText(name)
-
-                    # Emit signal when model is updated
-                    self.model.ModelRefreshed.emit()
-                    break
+            # Look up stored index to ID column
+            id_index = self.model_ids[file_id]
+            if not id_index.isValid():
+                return
+
+            # Update thumb for file
+            thumb_index = id_index.sibling(id_index.row(), 0, Qt.DecorationRole)
+            thumb_path = self.get_thumb_path(file_id, 1, clear_cache=True)
+            item = m.itemFromIndex(thumb_index)
+            item.setIcon(QIcon(thumb_path))
+            item.setText(name)
+
+            # Emit signal when model is updated
+            self.ModelRefreshed.emit()
+
         self.ignore_updates = False
 
+    def selected_file_ids(self):
+        """ Get a list of file IDs for all selected files """
+        # Get the indexes for column 5 of all selected rows
+        selected = self.selection_model.selectedRows(5)
+
+        return [idx.data() for idx in selected]
+
+    def selected_files(self):
+        """ Get a list of File objects representing the current selection """
+        files = []
+        for id in self.selected_file_ids():
+            files.append(File.get(id=id))
+        return files
+
+    def current_file_id(self):
+        """ Get the file ID of the current files-view item, or the first selection """
+        cur = self.selection_model.currentIndex()
+
+        if not cur or not cur.isValid() and self.selection_model.hasSelection():
+            cur = self.selection_model.selectedIndexes()[0]
+
+        if cur and cur.isValid():
+            return cur.sibling(cur.row(), 5).data()
+
+    def current_file(self):
+        """ Get the File object for the current files-view item, or the first selection """
+        cur_id = self.current_file_id()
```

Comment:
Same question, what if this doesn't get a value returned? And what if current_file() doesn't return a File object?